src/main/java/com/thebookcooper/controller/BookInfoController.java

```diff
@@ -82,4 +82,44 @@ public Book getBookById(@PathVariable("id") long id) {
         }
         return null; // Or return an appropriate response/entity indicating not found or error
     }
+
+    @PostMapping("/books/update/{id}")
+    public Book updateBook(@PathVariable("id") long id, @RequestBody String json) throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map inputMap = objectMapper.readValue(json, Map.class);
+
+        Book updatedBook = new Book();
+        try {
+            Connection connection = dcm.getConnection();
+            BookInfoDAO infoDAO = new BookInfoDAO(connection);
+
+            //get inputs from user and assign them to a new object
+            updatedBook.setBookId(id);
+            updatedBook.setTitle((String) inputMap.get("title"));
+            updatedBook.setISBN((int) inputMap.get("isbn"));
+            updatedBook.setPublishDate(Date.valueOf((String) inputMap.get("publishDate"))); //date has to be of form "YYYY-MM-DD"
+            updatedBook.setAuthor((String) inputMap.get("author"));
+            updatedBook.setGenre((String) inputMap.get("genre"));
+            updatedBook.setBookCondition((String) inputMap.get("bookCondition"));
+            updatedBook.setPrice((double) inputMap.get("price"));
+
+            //calls update function from dao/BookInfoDAO to update listing
+            return infoDAO.update(updatedBook);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new RuntimeException("Failed to update the book", e);
+        }
+    }
+
+    @GetMapping("/books/delete/{id}")
```

Comment:
having trouble with deleting a book if it's being used in the listings check it out:

ERROR: update or delete on table "book_info" violates foreign key constraint "book_listings_book_id_fkey" on table "book_listings
app-1             |   Detail: Key (book_id)=(1) is still referenced from table "book_listings".] with root cause                                                                                                                           app-1             |                                                                                                                                                                                                                        app-1             | org.postgresql.util.PSQLException: ERROR: update or delete on table "book_info" violates foreign key constraint "book_listings_book_id_fkey" on table "book_listings"                                                  
app-1             |   Detail: Key (book_id)=(1) is still referenced from table "book_listings".   

I know that listings is not on this current branch, but will this be a problem when it is? Do we need to delete the listing if the specific book id is removed too? Probably right?